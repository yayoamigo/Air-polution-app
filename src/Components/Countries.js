import React from 'react';
import { Link } from 'react-router-dom';

const Countries = ({name,aqi}) => {
  return (
    <div className='countries' key={name}>
      <h2>{name}</h2>
      <p>Air quality :{aqi}</p>
      <Link to={`/country/${name}`}  >
      <button>more...</button>
      </Link>
    </div>
  )
}

export default Countries;