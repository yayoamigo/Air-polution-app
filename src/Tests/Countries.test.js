import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Countries from '../Components/Countries';
import store from '../redux/configureStore';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';


it('render', () => {
  const tree = renderer
    .create(
    <Router>
        <Provider store={store}>
          <Countries name='Niger' aqi='2' />
        </Provider>
    </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders content', () => {
  const component = render(
    <Router>
    <Provider store={store}>
      <Countries name='Niger' aqi='2' />
    </Provider>
    </Router>
  );
  expect(component.container).toHaveTextContent('NigerAir quality :2more...');
});