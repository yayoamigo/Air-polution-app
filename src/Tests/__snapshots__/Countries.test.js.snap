// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="countries"
>
  <h2>
    Niger
  </h2>
  <p>
    Air quality :
    2
  </p>
  <a
    href="/country/Niger"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      more...
    </button>
  </a>
</div>
`;
